### PR TITLE
Fix initial sizing and tests.

### PR DESCRIPTION
**Pull request by pkaske**
Absolute fitting caused chart to not show up without explicit width and height.
Tests weren't working with the new API of Chart.js. Also updated chart options in the tests. Afaik all charts now use the same dataset based data structure, so I removed the basic-data-chart.html test file. It tested basically the same stuff. Curious if you guys agree?

### CODE REVIEW

File: test/basic-dataset-chart.html [hunk 1]
```diff
@@ -63,9 +61,9 @@
           chartEl.data = data;
           setTimeout(function() {
             expect(chartEl.chart).to.not.be.undefined;
-            expect(chartEl.chart.scale.height).to.be.above(0);
+            expect(chartEl.chart.scales['x-axis-0'].width).to.be.above(0);
             done();
-          }, 10);
```
Comment (pkaske) [inline test/basic-dataset-chart.html]: With a timeout of 10 I got sometimes flaky test results.

Comment (robdodson) [inline test/basic-dataset-chart.html]: Do you think it's because the chart is animating?

Comment (pkaske) [inline test/basic-dataset-chart.html]: Probably. I just took another look at the docs and found that a `onComplete` callback can be configured that is fired as soon as the chart finished it's animation. Maybe that a better route. http://www.chartjs.org/docs/#chart-configuration-animation-configuration

**Comment by robdodson** [comment]
oh yeah that seems like it could be a way better approach. Are you able to update the PR to use that?

**Comment by pkaske** [comment]
Yep, will do that. But not today. (It's 11pm in Germany)

**Comment by pkaske** [comment]
Now using onComplete to finish the tests. Works like a charm in my testing. But as said, I don't own a saucelaps account.

**Comment by robdodson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by pkaske** [comment]
At your service if you have questions, critic, recommendations, my master 😉

**Comment by robdodson** [comment]
@pkaske hah thanks :) Just trying to get my sauce account working after having neglected it for so long!

**Comment by robdodson** [comment]
Local tests work. Not sure what's up with Sauce Labs but we can deal with that another day :D

**Pull request merged**